edb/schema/utils.py

```diff
@@ -626,6 +626,22 @@ def get_class_nearest_common_ancestor(
         *[set(c.get_ancestors(schema).objects(schema)) | {c}
           for c in classes[1:]])
     common_list = sorted(common, key=lambda i: first.index(i))
+    nearests: List[so.InheritingObjectT] = []
+    # Then find the common ancestors that don't have any subclasses that
+    # are also nearest common ancestors.
+    for anc in common_list:
+        if not any(x.issubclass(schema, anc) for x in nearests):
+            nearests.append(anc)
+
+    return nearests
+
+
+def get_class_nearest_common_ancestor(
+    schema: s_schema.Schema,
+    classes: Iterable[so.InheritingObjectT]
+) -> Optional[so.InheritingObjectT]:
+    common_list = get_class_nearest_common_ancestors(schema, classes)
+    # FIXME: This is arbitrary and probably not meaningful
```

Comment:
Perhaps we could add a check to see if combining `classes` results in a meaningful MRO?  For example,

```
type A
type B
type C extending B, A
type D extending A, B

get_class_nearest_common_ancestor([C, D]) raises
```

Reply:
My plan was to go through and audit the call sites (of which there are only a single digit number) and try to work out what the best approach for each case is, but until I do that preserve the current behavior